File: app/src/main/java/com/smith/ufc/data/base/Repository.java
package com.smith.ufc.data.base;

import java.util.List;

/**
 * Created by Charlton on 10/31/17.
 */

public interface Repository<T, S> {
    void add(T item);

    void add(List<T> items);

    void update(T item);

    void remove(T item);

    List<T> query(S specification);
}

File: app/src/main/java/com/smith/ufc/data/base/Repository.kt
package com.smith.ufc.data.base

/**
 * Created by Charlton on 10/31/17.
 */

interface Repository<T, S> {
    fun add(item: T)

    fun add(items: List<T>)

    fun update(item: T)

    fun remove(item: T)

    fun query(specification: S): List<T>
}